src/main/java/edu/ucsb/cs56/mapache_search/AuthControllerAdvice.java

```diff
@@ -12,37 +12,39 @@
 
 import java.util.List;
 
-@ControllerAdvice //All ModelAttributes are shared to any view and this code runs before every controller action
+@ControllerAdvice // All ModelAttributes are shared to any view and this code runs before every
+                  // controller action
 public class AuthControllerAdvice {
 
-    @Autowired //it will find GithubOrgMembershipService and create bean/instance of it
+    @Autowired // it will find GithubOrgMembershipService and create bean/instance of it
     private MembershipService membershipService;
 
-    @Autowired //create instance of this
+    @Autowired // create instance of this
     private UserRepository userRepository;
 
     @ModelAttribute("isLoggedIn")
-    public boolean getIsLoggedIn(OAuth2AuthenticationToken token){
+    public boolean getIsLoggedIn(OAuth2AuthenticationToken token) {
         return token != null;
     }
 
     @ModelAttribute("id")
-    public String getUid(OAuth2AuthenticationToken token){
-        if (token == null) return "";
+    public String getUid(OAuth2AuthenticationToken token) {
+        if (token == null)
+            return "";
 
         String uid = token.getPrincipal().getAttributes().get("id").toString();
 
         List<AppUser> users = userRepository.findByUid(uid);
 
-        if (users.size()==0) {
+        if (users.size() == 0) {
             AppUser u = new AppUser();
             u.setUid(uid);
             u.setUsername(token2login(token));
             u.setApikey("");
+            u.setMaxsearches(20l);
```

Comment:
I'm still not sure what the 20 is doing here on line 44.

It seems more as if you should have, in AppUser:

```
public static final long MAX_SEARCHES=100l;
```

instead of a redundant instance variable `maxsearches` for 
each instance.

And then you should use 
```
u.setMaxSearches(AppUser.MAX_SEARCHES);
```

Unless I am misunderstanding your intentions?